File: examples/network_scripting.lua
-- 网络功能脚本示例
-- 演示如何在Lua脚本中使用游戏引擎的网络功能

-- 1. TCP客户端连接
local function tcp_example()
    -- 连接到服务器
    local result = tcp_connect("client1", "localhost", 8080)
    print("TCP连接结果:", result)

    -- 发送数据
    local send_result = tcp_send("client1", "Hello Server!")
    print("TCP发送结果:", send_result)
end

-- 2. WebSocket连接
local function websocket_example()
    -- 连接到WebSocket服务器
    local result = ws_connect("ws_client1", "ws://localhost:8080/ws")
    print("WebSocket连接结果:", result)
end

-- 3. HTTP请求
local function http_example()
    -- 发送GET请求
    local result = http_get("https://api.example.com/data")
    print("HTTP GET结果:", result)
end

-- 主函数
function main()
    print("=== 游戏引擎网络API示例 ===")

    -- 执行TCP示例
    tcp_example()

    -- 执行WebSocket示例
    websocket_example()

    -- 执行HTTP示例
    http_example()
end

-- 运行主函数
main()

File: examples/network_scripting.js
// 网络功能JavaScript示例
// 演示如何在JavaScript脚本中使用游戏引擎的网络功能

// 1. TCP客户端连接
async function tcpExample() {
    console.log("=== TCP客户端示例 ===");

    // 连接到服务器
    const result = await tcp_connect("client1", "localhost", 8080);
    console.log("TCP连接结果:", result);

    // 发送数据
    const sendResult = await tcp_send("client1", "Hello Server!");
    console.log("TCP发送结果:", sendResult);
}

// 2. WebSocket连接
async function websocketExample() {
    console.log("=== WebSocket示例 ===");

    // 连接到WebSocket服务器
    const result = await ws_connect("ws_client1", "ws://localhost:8080/ws");
    console.log("WebSocket连接结果:", result);
}

// 3. HTTP请求
async function httpExample() {
    console.log("=== HTTP请求示例 ===");

    // 发送GET请求
    const result = await http_get("https://api.example.com/data");
    console.log("HTTP GET结果:", result);
}

// 主函数
async function main() {
    console.log("=== 游戏引擎网络API示例 ===");

    // 执行TCP示例
    await tcpExample();

    // 执行WebSocket示例
    await websocketExample();

    // 执行HTTP示例
    await httpExample();
}

// 运行主函数
main().catch(console.error);
